api/types/databaseobjectimportrule/databaseobjectimportrule_test.go

```diff
@@ -0,0 +1,172 @@
+// Copyright 2024 Gravitational, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package databaseobjectimportrule
+
+import (
+	"testing"
+
+	"github.com/gravitational/trace"
+	"github.com/stretchr/testify/require"
+
+	"github.com/gravitational/teleport/api/defaults"
+	dbobjectimportrulev1 "github.com/gravitational/teleport/api/gen/proto/go/teleport/dbobjectimportrule/v1"
+	headerv1 "github.com/gravitational/teleport/api/gen/proto/go/teleport/header/v1"
+	"github.com/gravitational/teleport/api/types"
+)
+
+func TestNewDatabaseObjectImportRule(t *testing.T) {
+	tests := []struct {
+		name          string
+		spec          *dbobjectimportrulev1.DatabaseObjectImportRuleSpec
+		expectedError error
+	}{
+		{
+			name: "valid rule",
+			spec: &dbobjectimportrulev1.DatabaseObjectImportRuleSpec{
+				DbLabels: map[string]string{"key": "value"},
+				Mappings: []*dbobjectimportrulev1.DatabaseObjectImportRuleMapping{{}},
+			},
+			expectedError: nil,
+		},
+		{
+			name:          "nil spec",
+			spec:          nil,
+			expectedError: trace.BadParameter("missing spec"),
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			_, err := NewDatabaseObjectImportRule(tt.name, tt.spec)
+			require.ErrorIs(t, err, tt.expectedError)
```

Comment:
Please avoid ErrorIs.

Reply:
I think it's fine here, see other comment.